central/delegatedregistryconfig/delegator/delegator_test.go

```diff
@@ -0,0 +1,312 @@
+package delegator
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	deleDSMocks "github.com/stackrox/rox/central/delegatedregistryconfig/datastore/mocks"
+	connMocks "github.com/stackrox/rox/central/sensor/service/connection/mocks"
+	"github.com/stackrox/rox/generated/storage"
+	waiterMocks "github.com/stackrox/rox/pkg/waiter/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	ctxBG         = context.Background()
+	errBroken     = errors.New("broken")
+	fakeClusterID = "fake-cluster-id"
+	fakeWaiterID  = "fake-waiter-id"
+
+	fakeRegistry      = "fake-reg"
+	fakeImageFullName = fmt.Sprintf("%v/repo/something", fakeRegistry)
+	fakeImgName       = &storage.ImageName{FullName: fakeImageFullName}
+)
+
+func TestGetDelegateClusterID(t *testing.T) {
+	var deleClusterDS *deleDSMocks.MockDataStore
+	var connMgr *connMocks.MockManager
+	var waiterMgr *waiterMocks.MockManager[*storage.Image]
+	var d *delegatorImpl
+
+	fakeConnWithCap := connMocks.NewMockSensorConnection(gomock.NewController(t))
+	fakeConnWithCap.EXPECT().HasCapability(gomock.Any()).Return(true).AnyTimes()
+
+	fakeConnWithoutCap := connMocks.NewMockSensorConnection(gomock.NewController(t))
+	fakeConnWithoutCap.EXPECT().HasCapability(gomock.Any()).Return(false).AnyTimes()
+
+	setup := func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		connMgr = connMocks.NewMockManager(ctrl)
+		waiterMgr = waiterMocks.NewMockManager[*storage.Image](ctrl)
+		deleClusterDS = deleDSMocks.NewMockDataStore(ctrl)
+		d = New(deleClusterDS, connMgr, waiterMgr)
+	}
+
+	t.Run("error get config", func(t *testing.T) {
```

Comment:
A nit that doesn't have to change now, but this is a good example of when using a test suite makes sense. The setup is encapsulated in a function called SetupTest and you don't have to call setup each time, then each individual test is broken out as a TestXYZ function and the tests are individually runnable

Reply:
Yup! I had started with that, running a single test from the suite in isolation was not working well in `vscode` with `testify` and required dropping to the CLI which was cumbersome. Happy to move it back to the suite in the future.